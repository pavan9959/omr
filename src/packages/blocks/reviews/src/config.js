Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.httpGetMethod = "GET";
exports.httpPutMethod = "PUT";
exports.httpPostMethod = "POST";
exports.httpDeleteMethod = "DELETE";

exports.ReviewsApiContentType = "application/json";
exports.getReviewsAPiEndPoint = "reviews/reviews";

exports.errorTitle = "Error";
exports.errorAllFieldsAreMandatory = "All fields are mandatory.";

exports.titleTitle = "Title";
exports.descriptionTitle = "Description";
exports.accountIdTitle = "Account Id";
exports.anonymousTitle = "Anonymous";

exports.description = "Description";
exports.account_id = 1;
exports.anonymous = false;
exports.namePlaceHolder = "Name";
exports.titlePlaceHolder = "Title";
exports.numberPlaceHolder = "Phone number";
exports.descriptionPlaceHolder = "Description";

exports.labelTitleText = "Reviews";
exports.labelBodyText = "Reviews Body";

exports.btnExampleTitle = "CLICK ME";
// Customizable Area End