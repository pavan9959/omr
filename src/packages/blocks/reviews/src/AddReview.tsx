import React from "react";

// Customizable Area Start
import {
  StyleSheet,
  Text,
  ScrollView,
  TouchableOpacity,
  View,
  TextInput,
  TouchableWithoutFeedback,
  Platform,
  KeyboardAvoidingView
} from "react-native";
// Customizable Area End

import ReviewsController, { Props, configJSON } from "./ReviewsController";

export default class AddReview extends ReviewsController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  render() {
    return (
      <KeyboardAvoidingView
        behavior={this.isPlatformiOS() ? "padding" : undefined}
        style={styles.keyboardPadding}
      >
        <ScrollView keyboardShouldPersistTaps="always" style={styles.container}>
          <TouchableWithoutFeedback
            testID={"Background"}
            onPress={() => {
              this.hideKeyboard();
            }}
          >
            <View>
              <View style={styles.bgInputContainer}>
                <Text style={styles.titleInput}>{configJSON.titleTitle}</Text>
                <TextInput
                  testID="title" //Merge Engine::From BDS
                  style={styles.bgMobileInput} //UI Engine::From Sketch
                  placeholder={configJSON.titlePlaceHolder} //UI Engine::From Sketch
                  {...this.txtTitleProps} //Merge Engine::From BDS - {...this.testIDProps}
                />
              </View>
              <Text style={styles.titleInput}>{configJSON.descriptionTitle}</Text>
              <TextInput
                placeholder={configJSON.descriptionPlaceHolder}
                multiline
                testID="txtReviews" //Merge Engine::From BDS
                style={styles.bgMobileInputMessage} //UI Engine::From Sketch
                {...this.txtDescriptionProps} //Merge Engine::From BDS - {...this.testIDProps}
              />
              <TouchableOpacity
                style={styles.viewBtn}
                testID="btnSubmit" //Merge Engine::From BDS
                onPress={() => {
                  this.addQueryApi();
                }}
              >
                <Text style={styles.viewBtnText}>Add New Review</Text>
              </TouchableOpacity>
            </View>
          </TouchableWithoutFeedback>
        </ScrollView>
      </KeyboardAvoidingView>
    );
  }
}
// Customizable Area End

// Customizable Area Start
const styles = StyleSheet.create({
  keyboardPadding: { flex: 1 },
  container: {
    flex: 1,
    padding: 16,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    backgroundColor: "#fff"
  },
  goBack: {
    marginLeft: 16
  },
  viewBtn: {
    backgroundColor: "blue",
    paddingVertical: 8,
    borderRadius: 4,
    marginTop: 10,
    borderWidth: 1,
    borderColor: "blue"
  },
  viewBtnText: {
    color: "#fff",
    textAlign: "center",
    fontSize: 16
  },
  titleInput: {
    color: "#000"
  },
  bgInputContainer: {
    marginBottom: 25
  },
  bgMobileInput: {
    height: 45,
    borderBottomWidth: Platform.OS === "web" ? 0 : 1,
    borderColor: "#c9c9c9"
  },
  bgMobileInputMessage: {
    borderWidth: Platform.OS === "web" ? 0 : 1,
    borderColor: "#c9c9c9",
    height: 100,
    borderRadius: 15,
    paddingHorizontal: 10,
    paddingVertical: 20,
    marginTop: 20
  }
});
// Customizable Area End