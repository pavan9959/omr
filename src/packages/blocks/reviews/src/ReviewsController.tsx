import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  token: string;
  title: string;
  description: string;
  account_id: number;
  anonymous: boolean;
  ReviewsList: [];
  activeCreatedAt: string;
  isVisible: boolean;
  accountReview: any;
  // Customizable Area End
}

interface SS {
  id: any;
}

export default class ReviewsController extends BlockComponent<Props, S, SS> {
  reviewsApiCallId: any;
  deleteReviewApiCallId: any;
  addReviewApiCallId: any;
  updateReviewApiCallId: any;
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [
      getName(MessageEnum.SessionResponseMessage),
      getName(MessageEnum.RestAPIResponceMessage)
    ];

    this.state = {
      title: "",
      description: "",
      account_id: 1,
      anonymous: false,
      token: "",
      ReviewsList: [],
      accountReview: null,
      activeCreatedAt: "",
      isVisible: false
    };

    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async componentDidMount() {
    super.componentDidMount();
    this.getToken();
    if (this.isPlatformWeb() === false) {
      this.props.navigation.addListener("willFocus", () => {
        this.getToken();
      });
    }
    // Customizable Area Start
    // Customizable Area End
  }

  getToken = () => {
    const msg: Message = new Message(
      getName(MessageEnum.SessionRequestMessage)
    );
    this.send(msg);
  };

  async receive(from: string, message: Message) {
    // Customizable Area Start
    runEngine.debugLog("Message Recived", message);

    if (getName(MessageEnum.SessionResponseMessage) === message.id) {
      let token = message.getData(getName(MessageEnum.SessionResponseToken));
      this.setState({ token });
      this.getReviewsList(token);
    } else if (getName(MessageEnum.RestAPIResponceMessage) === message.id) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      var responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      runEngine.debugLog("API Message Recived", message);

      if (responseJson && responseJson.data) {
        if (apiRequestCallId === this.reviewsApiCallId) {
          console.log("reviews list",JSON.stringify(responseJson));
          this.setState({ accountReview: responseJson.data });
          let reviewsList = responseJson;
        }
        if (apiRequestCallId === this.addReviewApiCallId || apiRequestCallId === this.updateReviewApiCallId) {
          console.log("add review" + responseJson.data.id);
          if(responseJson.data.id === null) {
            this.updateQueryApi();
          } else  {
            this.props.navigation.goBack();
          }
        }
      } else if (responseJson && responseJson.errors) {
          if (responseJson.errors ) {
            if (apiRequestCallId === this.updateReviewApiCallId || apiRequestCallId === this.addReviewApiCallId) {
              responseJson.errors.forEach((error:any)=> {
                if ( error["token"] ) {
                  this.showAlert(configJSON.errorTitle, error["token"]);
                  this.props.navigation.goBack();
                } else if ( error["review"] ) {
                  this.showAlert(configJSON.errorTitle, error["review"].join("."));
                  this.props.navigation.goBack();
                } else {
                  this.showAlert(configJSON.errorTitle, error);
                }
              })
            }
          }
      }
    }
    // Customizable Area End
  }

  // Customizable Area Start
  txtNameProps = {
    onChangeText: (text: string) => {
      this.setState({ title: text });

      //@ts-ignore
      this.txtNameProps.value = text;
    }
  };

  txtTitleProps = {
    onChangeText: (text: string) => {
      this.setState({ title: text });

      //@ts-ignore
      this.txtTitleProps.value = text;
    }
  };
  submitProps = {
    onPress : () => {
      this.addQueryApi();
    }
  }
 
  txtDescriptionProps = {
    onChangeText: (text: string) => {
      this.setState({ description: text });
      //@ts-ignore
      this.txtDescriptionProps.value = text;
    }
  };

  addQuery = () => {
    this.props.navigation.navigate("AddReview");
  };

  hideModal = () => {
    this.setState({ isVisible: !this.state.isVisible });
  };

  setModal = (item: any) => {
    this.setState({
      title: item.attributes.title,
      account_id: item.account_id,
      description: item.attributes.description,
      accountReview: null,
      anonymous: item.attributes.anonymous,
      isVisible: !this.state.isVisible
    });
  };

  isStringNullOrBlank(str: string) {
    return str === null || str.length === 0;
  }

  updateQueryApi = () => {
    console.log("update");
    let data = {
      data: {
        attributes: {
          title: this.state.title,
          description: this.state.description,
          account_id: this.state.account_id,
          anonymous: this.state.anonymous
        }
      }
    };
    
    const header = {
      "Content-Type": configJSON.ReviewsApiContentType,
      token: this.state.token
    };
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.updateReviewApiCallId = requestMessage.messageId;

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.getReviewsAPiEndPoint+"/"+this.state.account_id
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      JSON.stringify(data)
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.httpPutMethod
    );
    runEngine.sendMessage(requestMessage.id, requestMessage);
  }
  addQueryApi = () => {
      let data = {
        data: {
          attributes: {
            title: this.state.title,
            description: this.state.description,
            account_id: this.state.account_id,
            anonymous: this.state.anonymous
          }
        }
      };
      
      const header = {
        "Content-Type": configJSON.ReviewsApiContentType,
        token: this.state.token
      };
      const requestMessage = new Message(
        getName(MessageEnum.RestAPIRequestMessage)
      );

      this.addReviewApiCallId = requestMessage.messageId;

      requestMessage.addData(
        getName(MessageEnum.RestAPIResponceEndPointMessage),
        configJSON.getReviewsAPiEndPoint
      );
      requestMessage.addData(
        getName(MessageEnum.RestAPIRequestHeaderMessage),
        JSON.stringify(header)
      );
      requestMessage.addData(
        getName(MessageEnum.RestAPIRequestBodyMessage),
        JSON.stringify(data)
      );
      requestMessage.addData(
        getName(MessageEnum.RestAPIRequestMethodMessage),
        configJSON.httpPostMethod
      );
      runEngine.sendMessage(requestMessage.id, requestMessage);
      //@ts-ignore
      this.state.ReviewsList.push({attributes: data.attributes});
  };

  getReviewsList = (token: string) => {
    const header = {
      "Content-Type": configJSON.ReviewsApiContentType,
      token: token
    };
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.reviewsApiCallId = requestMessage.messageId;

    console.log("reviews list", this.state.ReviewsList);
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.getReviewsAPiEndPoint + "?account_id="+ this.state.account_id
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.httpGetMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);
  };
  // Customizable Area End
}
