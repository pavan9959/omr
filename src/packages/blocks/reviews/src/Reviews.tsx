import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  ScrollView,
  TouchableOpacity,
  View,
  Modal,
  Platform
} from "react-native";
// Customizable Area End

import ReviewsController, { Props } from "./ReviewsController";

export default class Reviews extends ReviewsController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  render() {
    return (
      <ScrollView style={styles.container}>
      {/* Customizable Area Start */}
        <View style={this.state.isVisible ? styles.modalBox : {display: "none"}}>
          <Modal
            animationType={"fade"}
            transparent={false}
            visible={this.state.isVisible}
          >
            <View style={styles.modal}>
              <ScrollView>
                <Text style={styles.infoText}>
                  <Text style={styles.labelText}>Title:{"  "}</Text>
                  {this.state.title}
                </Text>
                <Text numberOfLines={2} style={styles.infoText}>
                  <Text style={styles.labelText}>Anonymous{"  "}</Text>
                  {this.state.anonymous}
                </Text>
                <Text numberOfLines={2} style={styles.infoText}>
                  <Text style={styles.labelText}>Account ID{"  "}</Text>
                  {this.state.account_id}
                </Text>
                <Text numberOfLines={10} style={styles.infoText}>
                  <Text style={styles.labelText}>Description:{"  "}</Text>
                  {this.state.description}
                </Text>
                <Text style={styles.infoText}>
                  <Text style={styles.labelText}>Created At:{"  "}</Text>
                  {this.state.activeCreatedAt}
                </Text>
              </ScrollView>
              <View style={styles.buttonBox}>
                <TouchableOpacity
                  style={[styles.viewBtnWidth, styles.closeBtn]}
                  onPress={() => {
                    this.hideModal();
                  }}
                >
                  <Text style={styles.closeBtnText}>Close</Text>
                </TouchableOpacity>
              </View>
            </View>
          </Modal>
        </View>
        <TouchableOpacity
          testID={"btnAddReview"}
          style={this.state.isVisible ? {display: "none"} : styles.addBtn}
          onPress={() => {
            this.addQuery();
          }}
        >
          <Text style={styles.addtext}>Add New Review</Text>
        </TouchableOpacity>
        {!this.state.isVisible && this.state.accountReview?.attributes ? 
        (
          <View style={styles.tableBox}>
          <Text>ID: {this.state.accountReview.attributes.id}</Text>
          <Text>title: {this.state.accountReview.attributes.title}</Text>
          <Text>description: {this.state.accountReview.attributes.description}</Text>
          <Text>anonymous: {this.state.accountReview.attributes.anonymous ? "true" : "false"} </Text>
          </View>
        )
        :
        (<Text>{this.state.token && this.state.token.length > 0 ? "Cannot Find Review for Account " + this.state.account_id : "Please login for reviews."} </Text>)
        }
      {/* Customizable End Start */}
      </ScrollView>
      //Merge Engine End DefaultContainer
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    backgroundColor: "#ffffffff"
  },
  tableBox: {
    borderWidth: 1,
    borderRadius: 4,
    borderColor: "#ccc",
    padding: 15,
    marginVertical: 10,
    marginBottom: 30
  },
  webviewStyle: {
    height: 200
  },
  infoText: {
    fontSize: 16,
    marginVertical: 4
  },
  labelText: {
    fontWeight: "bold"
  },
  viewBtn: {
    backgroundColor: "blue",
    paddingVertical: 8,
    borderRadius: 4,
    marginTop: 10,
    borderWidth: 1,
    borderColor: "blue"
  },
  viewBtnText: {
    color: "#fff",
    textAlign: "center",
    fontSize: 16
  },
  addBtn: {
    backgroundColor: "blue",
    marginBottom: 10,
    width: 120,
    height: 40,
    display: "flex",
    justifyContent: "center",
    borderRadius: 4,
    alignSelf: "flex-end"
  },
  addtext: {
    fontSize: 16,
    color: "#fff",
    textAlign: "center"
  },
  buttonTop: {
    display: "flex",
    alignSelf: "flex-end",
    alignItems: "flex-end",
    justifyContent: "flex-end"
  },
  closeBtn: {
    backgroundColor: "#fff",
    paddingVertical: 8,
    borderRadius: 4,
    marginTop: 10,
    borderColor: "#ccc",
    borderWidth: 1
  },
  buttonBox: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between"
  },
  modalBox: {
    alignItems: "center",
    justifyContent: "center",
    flex: 1,
    display: "flex",
    width: "100%",
    height: "100%",
    backgroundColor: "#fff"
  },
  modal: {
    width: "80%",
    borderRadius: 4,
    borderWidth: 1,
    borderColor: "#ccc",
    marginTop: 80,
    marginLeft: 40,
    padding: 15
  },
  viewBtnWidth: {
    width: "48%"
  },
  closeBtnText: {
    color: "#000",
    textAlign: "center",
    fontSize: 16
  }
});
// Customizable Area End
