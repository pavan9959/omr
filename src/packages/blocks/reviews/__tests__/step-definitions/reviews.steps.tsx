import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import Reviews from "../../src/Reviews";
import AddReviews from "../../src/AddReview"
const navigation = require("react-navigation");

import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";
export const configJSON = require("../../config.json");
import MessageEnum, {
  getName
} from "../../../../framework/src/Messages/MessageEnum";
import { _ } from "../../../../framework/src/IBlock";

const screenProps = {
  navigation: navigation,
  id: "Reviews"
};

const feature = loadFeature("./__tests__/features/Reviews-scenario.feature");

defineFeature(feature, test => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to Reviews", ({ given, when, then }) => {
    let ReviewsWrapper: ShallowWrapper;
    let instance: Reviews;

    given("I am a User loading Reviews", () => {
      ReviewsWrapper = shallow(<Reviews {...screenProps} />);
    });

    when("I navigate to the Reviews", () => {
  
      instance = ReviewsWrapper.instance() as Reviews;

      const tokenMsg: Message = new Message(getName(MessageEnum.SessionResponseMessage));
      tokenMsg.addData(getName(MessageEnum.SessionResponseToken), "TOKEN");
      runEngine.sendMessage("Unit Test", tokenMsg);

      const getReviewsAPI = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );

      getReviewsAPI.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        {data: [{"id":"10","type":"Review","attributes":{"name":"Tester","email":"test@me.com","phone_number":"13015551212","description":"None","created_at":"2021-03-08T23:17:49.068Z","user":"Firstname Lastname"}}]}
      );
      instance.reviewsApiCallId = getReviewsAPI.messageId;
      runEngine.sendMessage("Unit Test", getReviewsAPI);

      const deleteReviewsAPI = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      instance.deleteReviewApiCallId = deleteReviewsAPI.messageId;
      runEngine.sendMessage("Unit Test", deleteReviewsAPI);


      const createReviewsAPI = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      createReviewsAPI.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        {"errors":[{"Review":["Phone number invalid"]}]}
      );
      
      instance.reviewsApiCallId = createReviewsAPI.messageId;
      runEngine.sendMessage("Unit Test", createReviewsAPI);
     
    });




    then("Reviews will load with out errors", () => {
      expect(ReviewsWrapper).toBeTruthy();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(ReviewsWrapper).toBeTruthy();
    });
  });

  test("User navigates to addReviews", ({ given, when, then }) => {
    let AddReviewsWrapper: ShallowWrapper;
    let instance: AddReviews;

    given("I am a User loading addReviews", () => {
      AddReviewsWrapper =  shallow(<AddReviews {...screenProps} />);
    });

    when("I navigate to the addReviews", () => {
      instance = AddReviewsWrapper.instance() as AddReviews;

      let buttonComponent = AddReviewsWrapper.findWhere((node) => node.prop('testID') === 'btnSubmit');
      buttonComponent.simulate('press');

      let textInputComponent = AddReviewsWrapper.findWhere((node) => node.prop('testID') === 'txtReviews');
      textInputComponent.simulate('changeText', 'N/A');

      buttonComponent.simulate('press');

    });

    then("addReviews will load with out errors", () => {
      expect(AddReviewsWrapper).toBeTruthy();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(AddReviewsWrapper).toBeTruthy();
    });
  });

});
