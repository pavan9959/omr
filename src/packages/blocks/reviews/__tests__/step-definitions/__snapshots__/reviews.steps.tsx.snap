// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reviews User navigates to Reviews 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Component
      animationType="fade"
      hardwareAccelerated={false}
      transparent={false}
      visible={false}
    >
      <View
        style={
          Object {
            "borderColor": "#ccc",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginLeft": 40,
            "marginTop": 80,
            "padding": 15,
            "width": "80%",
          }
        }
      >
        <ScrollViewMock>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Title:
                
            </Text>
          </Text>
          <Text
            numberOfLines={2}
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Anonymous
                
            </Text>
          </Text>
          <Text
            numberOfLines={2}
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Account ID
                
            </Text>
            1
          </Text>
          <Text
            numberOfLines={10}
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Description:
                
            </Text>
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Created At:
                
            </Text>
          </Text>
        </ScrollViewMock>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "width": "48%",
                },
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#ccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginTop": 10,
                  "paddingVertical": 8,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              Close
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </Component>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "blue",
        "borderRadius": 4,
        "display": "flex",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": 120,
      }
    }
    testID="btnAddReview"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Add New Review
    </Text>
  </TouchableOpacity>
  <Text>
    Cannot Find Review for Account 1
     
  </Text>
</ScrollViewMock>
`;

exports[`reviews User navigates to Reviews 2`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Component
      animationType="fade"
      hardwareAccelerated={false}
      transparent={false}
      visible={false}
    >
      <View
        style={
          Object {
            "borderColor": "#ccc",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginLeft": 40,
            "marginTop": 80,
            "padding": 15,
            "width": "80%",
          }
        }
      >
        <ScrollViewMock>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Title:
                
            </Text>
          </Text>
          <Text
            numberOfLines={2}
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Anonymous
                
            </Text>
          </Text>
          <Text
            numberOfLines={2}
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Account ID
                
            </Text>
            1
          </Text>
          <Text
            numberOfLines={10}
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Description:
                
            </Text>
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Created At:
                
            </Text>
          </Text>
        </ScrollViewMock>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "width": "48%",
                },
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#ccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginTop": 10,
                  "paddingVertical": 8,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              Close
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </Component>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "blue",
        "borderRadius": 4,
        "display": "flex",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": 120,
      }
    }
    testID="btnAddReview"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Add New Review
    </Text>
  </TouchableOpacity>
  <Text>
    Cannot Find Review for Account 1
     
  </Text>
</ScrollViewMock>
`;

exports[`reviews User navigates to addReviews 1`] = `
<KeyboardAvoidingView
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    keyboardShouldPersistTaps="always"
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 650,
        "padding": 16,
        "width": "100%",
      }
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
      testID="Background"
    >
      <View>
        <View
          style={
            Object {
              "marginBottom": 25,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
              }
            }
          >
            Title
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Title"
            rejectResponderTermination={true}
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#c9c9c9",
                "height": 45,
              }
            }
            testID="title"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000",
            }
          }
        >
          Description
        </Text>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Description"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#c9c9c9",
              "borderRadius": 15,
              "borderWidth": 1,
              "height": 100,
              "marginTop": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 20,
            }
          }
          testID="txtReviews"
          underlineColorAndroid="transparent"
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderColor": "blue",
              "borderRadius": 4,
              "borderWidth": 1,
              "marginTop": 10,
              "paddingVertical": 8,
            }
          }
          testID="btnSubmit"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Add New Review
          </Text>
        </TouchableOpacity>
      </View>
    </TouchableWithoutFeedback>
  </ScrollViewMock>
</KeyboardAvoidingView>
`;

exports[`reviews User navigates to addReviews 2`] = `
<KeyboardAvoidingView
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    keyboardShouldPersistTaps="always"
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 650,
        "padding": 16,
        "width": "100%",
      }
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
      testID="Background"
    >
      <View>
        <View
          style={
            Object {
              "marginBottom": 25,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
              }
            }
          >
            Title
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Title"
            rejectResponderTermination={true}
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#c9c9c9",
                "height": 45,
              }
            }
            testID="title"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000",
            }
          }
        >
          Description
        </Text>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Description"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#c9c9c9",
              "borderRadius": 15,
              "borderWidth": 1,
              "height": 100,
              "marginTop": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 20,
            }
          }
          testID="txtReviews"
          underlineColorAndroid="transparent"
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderColor": "blue",
              "borderRadius": 4,
              "borderWidth": 1,
              "marginTop": 10,
              "paddingVertical": 8,
            }
          }
          testID="btnSubmit"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Add New Review
          </Text>
        </TouchableOpacity>
      </View>
    </TouchableWithoutFeedback>
  </ScrollViewMock>
</KeyboardAvoidingView>
`;
